experimental/emulated_vault/opt/emulated-vault/vault-server

```diff
@@ -0,0 +1,249 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+#
+
+import json
+import logging
+import sys
+import os
+import inspect
+from BaseHTTPServer import HTTPServer
+from BaseHTTPServer import BaseHTTPRequestHandler
+import ssl
+import urlparse, json
+import re, fnmatch
+
+import adaptor.fs
+
+class RequestHandler(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        try:
+            self._do_GET()
+        except:
+            logger.exception("do_GET exception")
+
+    do_HEAD = do_GET
+
+    def do_POST(self):
+        try:
+            self._do_POST()
+        except:
+            logger.exception("do_POST exception")
+
+    do_PUT = do_POST
+
+    def do_DELETE(self):
+        try:
+            self._do_DELETE()
+        except:
+            logger.exception("do_DELETE exception")
+
+    def _do_GET(self):
+        logger.debug("GET %s", self.path)
+        parsed_path = urlparse.urlparse(self.path)
+
+        if parsed_path.path == "/ping":
+            logger.info("Ping")
+            success, value = adaptor.ping()
+            if not success:
+                self.send_response(503)
+                self.send_header('Content-type', 'text/html')
+                self.end_headers()
+                self.wfile.write("Failure")
+                return
+
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html')
+            self.end_headers()
+            self.wfile.write(value)
+            return
+
+        if parsed_path.path == "/search/query/sslkeys":
+            logger.info("Search SSL: path=%s query=%s", parsed_path.path, parsed_path.query)
+            filters = {}
+            keyFilters = {}
+            cdnFind = re.search(".*q=cdn:([^&]*).*", parsed_path.query)
+            if cdnFind:
+                def cdnFilter(key,val):
+                    try:
+                        data = json.loads(val)
+                        return data['cdn']==cdnFind.group(1)
+                    except:
+                        return False
+                filters['cdn'] = cdnFilter
+
+            dsFind = re.search(".*q=deliveryservice:([^&]*).*", parsed_path.query)
+            if dsFind:
+                def dsFilter(key,val):
+                    try:
+                        data = json.loads(val)
+                        return data['deliveryservice']==dsFind.group(1)
+                    except:
+                        return False
+                filters['ds'] = dsFilter
+
+            keyFind = re.search(".*q=_yz_rk:([^&]*).*", parsed_path.query)
+            if keyFind:
+                def keyFilter(key):
+                    return fnmatch.fnmatch(os.path.basename(key), keyFind.group(1))
+                keyFilters['key-match'] = keyFilter
+
+            success, parameters = adaptor.searchParameters("riak/ssl/", keyFilters=keyFilters, filters=filters)
+            if not success:
+                self.send_response(503)
```

Comment:
is this the right response code for a failure of parameter search?

Reply:
Probably copied RIAK behavior